### PR TITLE
API to remove single file from PC

### LINKED ISSUE #1603: add API for "remove file from presentation compiler"
The Scala presentation compiler works by type-checking a set of files in the background. Files are added to the set of "managed" files by `askReload`.  During type-checking errors are reported _only_ for these files.

Usually, files are added to the "managed" set when they are open in a tab and removed when the tab is closed. The smaller the set of files that are managed, the faster the round-trip and the more responsive the PC is. Other than that, there is no harm in keeping files around.

The corresponding method for removing a file is [removeUnitOf](https://github.com/scala/scala/blob/2.12.x/src/interactive/scala/tools/nsc/interactive/CompilerControl.scala#L21). Currently Ensime does not offer an API for this call, only "removeAll". In the Eclipse implementation, this is surrounded by `ask`, so that's probably a good idea.

### PR DESCRIPTION
**Pull request by sugakandrey**
close #1603

### CODE REVIEW

File: api/src/main/scala/org/ensime/api/incoming.scala [hunk 1]
```diff
@@ -56,6 +56,11 @@ final case class TypecheckFileReq(fileInfo: SourceFileInfo) extends RpcAnalyserR
 final case class UnloadModuleReq(module: String) extends RpcAnalyserRequest
 
 /**
+ * Responds with a `VoidResponse`
+ */
+final case class UnloadFileReq(file: File) extends RpcAnalyserRequest
```
Comment (fommil) [inline api/src/main/scala/org/ensime/api/incoming.scala]: can we make this a `SourceFileInfo`? We're trying to deprecate uses of `File` in the API

File: protocol-swanky/src/main/scala/org/ensime/server/protocol/swank/SwankFormats.scala [hunk 2]
```diff
@@ -600,6 +600,7 @@ object SwankProtocolRequest {
   implicit val ConnectionInfoReqHint: TypeHint[ConnectionInfoReq.type] = TypeHint[ConnectionInfoReq.type](SexpSymbol("swank:connection-info"))
 
   implicit val RemoveFileReqHint: TypeHint[RemoveFileReq] = TypeHint[RemoveFileReq](SexpSymbol("swank:remove-file"))
+  implicit val UnloadFileReqHint: TypeHint[UnloadFileReq] = TypeHint[UnloadFileReq](SexpSymbol("swank:unload-file"))
```
Comment (fommil) [inline protocol-swanky/src/main/scala/org/ensime/server/protocol/swank/SwankFormats.scala]: I pity anyone who has to touch this file...

**Comment by fommil** [comment]
@dragos ^^^

File: api/src/main/scala/org/ensime/api/incoming.scala (hunk 1 above)
Comment (dragos) [inline api/src/main/scala/org/ensime/api/incoming.scala]: Would it make sense to use URIs to talk about file identity when the contents don't matter?

**Comment by dragos** [comment]
In what regards PC interaction, this looks good to me. Thank you for picking up this ticket, it's may look small but I think it will improve the usability quite a lot in the long term. At least, I plan to use it in VS Code :)

**Comment by fommil** [comment]
yeah, if I have the time adding it to emacs would be great too, e.g. on file close. Maybe save me reloading open files when things get weird.

**Comment by sugakandrey** [comment]
@fommil can't merge this until review is approved.

**Comment by fommil** [comment]
@dragos by the wonders of CI, you should get this if you restart your server in 30 mins (unless you're caching it)

**Pull request merged**